isisd/isis_spf.c

```diff
@@ -2190,136 +2369,352 @@ DEFUN(show_isis_topology, show_isis_topology_cmd,
 	const char *vrf_name = VRF_DEFAULT_NAME;
 	bool all_vrf = false;
 	int idx_vrf = 0;
+	uint8_t algorithm = 0;
 
-	if (argv_find(argv, argc, "topology", &idx)) {
-		if (argc < idx + 2)
-			levels = ISIS_LEVEL1 | ISIS_LEVEL2;
-		else if (strmatch(argv[idx + 1]->arg, "level-1"))
-			levels = ISIS_LEVEL1;
-		else
-			levels = ISIS_LEVEL2;
+#ifndef FABRICD
+	levels = ISIS_LEVEL1 | ISIS_LEVEL2;
+	if (argv_find(argv, argc, "level-1", &idx))
+		levels = ISIS_LEVEL1;
+	if (argv_find(argv, argc, "level-2", &idx))
+		levels = ISIS_LEVEL2;
+#endif
+	if (argv_find(argv, argc, "algorithm", &idx))
+		algorithm = (uint8_t)strtoul(argv[idx + 1]->arg, NULL, 10);
+
+	if (!im) {
+		vty_out(vty, "IS-IS Routing Process not enabled\n");
+		return CMD_SUCCESS;
+	}
```

Comment:
Flex Algo should be exclude from fabricd

Reply:
done